kb(gantt): different colors for tasks (#2121)
telerik/blazor-docs: knowledge-base/gantt-different-colors-for-tasks.md

---
title: Set Different Colors for the Gantt Tasks
description: How to set different colors for the Gantt tasks?
type: how-to
page_title: Set Different Colors for the Gantt Tasks
slug: gantt-kb-different-colors-for-tasks
position: 
tags: 
ticketid: 1536209
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Gantt for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

I want to set different colors for the Gantt tasks. I store the colors in my data source. How to apply the colors from my data source to the rendered Gantt tasks?

## Solution

Follow these steps to apply the desired colors to the Gantt tasks:

1. Use a [Task Template]({%slug gantt-task-template%}) to override the default task rendering.
1. In the template, declare a wrapping element that will contain all the task details. Specify the desired task information inside.

> 1. Add an attribute `style="background-color"` to the wrapper. Set the color from the template `context` or any other desired color.

Reviewer: Please, double-check the second sentence here as it sounds weird. It might be correct but it's a bit confusing.

Reviewer 2: > Please, double-check the second sentence here as it sounds weird. It might be correct but it's a bit confusing.

That's alright. It implies color can be set both from the context model passed to the template (e.g. predefined based on the task), or an abitrary color alltogether.

> 1. Use custom CSS remove the default padding of the task template element (`<div class="k-task-template">`) and adjust the custom wrapper, so it covers the whole task element. This will ensure that the background color will cover the whole task.

Reviewer: ```suggestion
1. Use custom CSS to remove the default padding of the task template element (`<div class="k-task-template">`) and adjust the custom wrapper, so it covers the whole task element. This ensures that the background color covers the whole task.
```

        }

        Data.Remove(item);
    }

    class FlatModel
    {
        public int Id { get; set; }
        public int? ParentId { get; set; }
        public string Title { get; set; }
        public double PercentComplete { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public string Assignee { get; set; }
        public string Color { get; set; }
    }
}

<style>
    .k-task-single {
        background-color: none;
    }

    .k-task-actions {
        position: absolute;
        right: 0;
        top: 14px;
    }

    .k-task-template {
        padding: 0px;
    }

    .k-task-content {
        display: block;
    }

    .wrapper {
        padding-top: 4px;
        padding-bottom: 4px;
        padding-left: 8px;
        padding-right: 8px;
        border-radius: 4px
    }

    .task-title {
        font-size: 18px;
        font-weight: bold;
        border-bottom: 1px solid;
    }

    .task-assignee {
        font-style: italic;
    }

    .progress {
        position: absolute;
        left: 0;
        bottom: 0;
        height: 4px;
        background: rgba(252, 98, 87, .7);
    }
</style>

> ````

Reviewer: Add a `## See Also` section with links to:

* Gantt overview
* Gantt live demo
* Gantt task template
* Gantt API

In addition, for SEO purposes you might want to figure out a Gantt page from which to link to this KB.
